parsing/pprintast.ml

```diff
@@ -1074,11 +1075,11 @@ and signature_item ctxt f x : unit =
         | pmd :: tl ->
             if not first then
               pp f "@ @[<hov2>and@ %s:@ %a@]%a" pmd.pmd_name.txt
-                (module_type ctxt) pmd.pmd_type
+                (module_type1 ctxt) pmd.pmd_type
```

Comment:
Why are you using the inner-level here and below? (And not in `Psig_module`.)

Reply:
Otherwise you would get
```ocaml
module rec A : S with type t = t and B : S with type t = t
```
which can't be parsed (the `and B : S` is considered part of the constraint). In principle, it's not really necessary below, but I did it for symmetry.

In `Psig_module` there is no `and` possible. Is it better to be consequent and add parentheses there too?